daxa/mission/erosita.py

```diff
@@ -0,0 +1,703 @@
+            gd_insts = [int(re.sub('[^0-9]','', tscope)) for tscope in insts]
+            
+            # Getting the indexes of events with the chosen insts
+            gd_insts_indx = np.hstack([(t_col==i).nonzero()[0] for i in gd_insts])
+            
+            # Filtering the data on those tscopes
+            filtered_data = data[gd_insts_indx]
+
+            # Replacing unfiltered eventlist in the fits file with the new ones
+            hdul[1].data = filtered_data
+
+            # Writing this to a new file (the if is for instrument filtered)
+            hdul.writeto(evlist_path[:-5] + '_if_{}.fits'.format(insts_str))
+            hdul.close()
+        
+    @staticmethod
+    def _download_call(self, field: str):
+        """
+        This internal static method is purely to enable parallelised downloads of data, as defining
+        an internal function within download causes issues with pickling for multiprocessing.
+
+        :param str field: The field name of the particular field to be downloaded.
+        :param str filename: The directory under which to save the downloaded tar.gz.
+        :return: A None value.
+        :rtype: Any
+        """
+        # Since you can't download a single observation for a field, you have to download them all in one tar file,
+        #  I am making a temporary directory to download the tar file and unpack it in, then move the observations 
+        #  to their own directory afterwards in the _directory_formatting function
+        temp_dir = os.path.join(self.raw_data_path, "temp_download")
+        if not os.path.exists(temp_dir):
+            os.makedirs(temp_dir)
+
+        # Download the requested data
+        r = requests.get(CalPV_info["download"].loc[CalPV_info["Field_Name"] == field].values[0])
+        field_dir = os.path.join(temp_dir, "{f}".format(f=field))
+        os.makedirs(field_dir)
+        open(field_dir + "{f}.tar.gz".format(f=field), "wb").write(r.content)
+        # unzipping the tar file
+        tarname = field_dir + "{f}.tar.gz".format(f=field)
+        tar = tarfile.open(tarname, "r:gz")
+        tar.extractall(field_dir)
+        tar.close()
+        os.remove(tarname)
+
+        return None
+    
+    def _directory_formatting(self):
+        """
+        Internal method to rearrange the downloaded files from field names into the Obs_ID top layer 
+        directory structure for consistency with other missions in DAXA. To be called after the initial 
+        download of the fields has been completed.
+        """
+        # Only executing the method if new data has been downloaded
+        if os.path.exists(os.path.join(self.raw_data_path, "temp_download")):
+            # Creating a dictionary that stores obs_ids as keys and their field name as values
+            field_dict = self._field_dict_generator()
+
+            # Moving the eventlist for each obs_id from its downloaded path to the path DAXA expects
+            for o in self.filtered_obs_ids:
```

Comment:
Informative variable names please. Single letter variables are to be avoided, apart from perhaps in list comprehensions. This could be 'for obs_id in self.filtered_obs_ids:'